### PR TITLE
preference syncing with SDK

### PR DESCRIPTION
**Pull request by seanyesmunt**
Also added a better titlebar for mac because I really wanted one

Most of the sync stuff was in different repos. The sync related changes here are in `store.js`

### CODE REVIEW

File: src/ui/util/saved-passwords.js [hunk 1]
```diff
@@ -11,10 +11,16 @@ export const setSavedPassword = value => {
...
   return new Promise(resolve => {
+    // @if TARGET='app'
     ipcRenderer.once('get-password-response', (event, password) => {
       resolve(password);
     });
     ipcRenderer.send('get-password');
+    // @endif
+
+    // @if TARGET='web'
+    resolve();
```
Comment (kauffj) [inline src/ui/util/saved-passwords.js]: I would make this `resolve('')` and add a comment explaining why the web doesn't need passwords

File: src/ui/store.js [hunk 2]
```diff
@@ -98,40 +105,54 @@ history = createHashHistory();
 history = createBrowserHistory();
 // @endif
 
+const triggerSharedStateActions = [
+  ACTIONS.CHANNEL_SUBSCRIBE,
```
[the remark this answers is not in the document]
Comment (seanyesmunt) [reply src/ui/store.js]: I don't think so, this allows apps to trigger the callback for app-specific sync data.

File: src/ui/store.js [hunk 3]
```diff
@@ -98,40 +105,54 @@ history = createHashHistory();
...
+    },
+  },
+  blocked: { source: 'blocked', property: 'blockedChannels' },
+};
+
+const sharedStateCb = ({ dispatch, getState }) => {
+  const state = getState();
+  const syncEnabled = makeSelectClientSetting(SETTINGS.ENABLE_SYNC)(state);
+  const emailVerified = selectUserVerifiedEmail(state);
+  if (syncEnabled && emailVerified) {
```
[the remark this answers is not in the document]
Comment (seanyesmunt) [reply src/ui/store.js]: Why would we sync if they don't have an email?

File: src/ui/store.js [hunk 4]
```diff
@@ -98,40 +105,54 @@ history = createHashHistory();
...
+  LBRY_REDUX_ACTIONS.TOGGLE_BLOCK_CHANNEL,
+  LBRY_REDUX_ACTIONS.CREATE_CHANNEL_COMPLETED,
+];
+
+/**
+ * source: the reducer name
+ * property: the property in the reducer-specific state
+ * transform: optional method to modify the value to be stored
+ */
+const sharedStateFilters = {
```
[the remark this answers is not in the document]
Comment (seanyesmunt) [reply src/ui/store.js]: I'll add a comment with a link to that code.

**Pull request merged**